PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java

```diff
@@ -365,9 +365,9 @@ public void run() {
                         FragmentManager fragmentManager = getSupportFragmentManager();
                         fragmentManager.beginTransaction()
                                 .replace(R.id.content_frame, frag)
-                                .addToBackStack(null)
+                                .addToBackStack("Main Activity")
                                 .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
-                                .commit();
+                                .commitAllowingStateLoss();
```

Comment:
Any changes to activities / back button should probably be in a separate pull request to avoid conflating 2 different issues when reviewing. Paging @jasontsh to look over